tools/grid/test/grid.test.ts

```diff
@@ -128,9 +134,43 @@ describe('Grid', () => {
         await nextFrame();
         await nextFrame();
 
-        expect(
-            el.querySelector(el.focusableSelector) === document.activeElement
-        ).to.be.true;
+        const firstItem = el.querySelector(el.focusableSelector) as HTMLElement;
+        expect(firstItem === document.activeElement).to.be.true;
+        expect(el.tabIndex).to.equal(-1);
+
+        await sendKeys({
+            press: 'Tab',
+        });
+
+        await nextFrame();
+        await nextFrame();
+
+        await elementUpdated(el);
+
+        let activeElement = document.activeElement as HTMLElement;
+        expect(firstItem.contains(activeElement)).to.be.true;
+        expect(activeElement.tagName).to.equal('SP-ACTION-MENU');
+        expect(activeElement.tabIndex).to.equal(-1);
+        expect(el.tabIndex).to.equal(-1);
+
+        await sendKeys({
+            press: 'Tab',
+        });
+
+        await nextFrame();
+        await nextFrame();
+        await nextFrame();
+        await nextFrame();
```

Comment:
Rather than relying on nextFrame, if we can connect the test to a certain thing we are waiting for, it ensures more stable test results.

Reply:
Replaced all the brittle nextFrame() calls with robust waitUntil() calls that wait for actual state changes.